Kafka Connect docs review (#932)
hazelcast/hz-docs: docs/modules/integrate/pages/kafka-connect-connectors.adoc

== Fault Tolerance

> When a Kafka Connect Source connector is deployed to a Hazelcast cluster, snapshots of the connector state are stored by the Jet engine.

Reviewer: Passive, recast to something like:

When you deploy a Kafka Connect Source connector to a Hazelcast cluster, the Jet Engine stores snapshots of the connector state.

> Example for such states are partition offsets and any metadata required for a restart or recovery. If there is a connector failure, the recovery behavior will depend on the type of connector. Refer to the documentation of your Kafka Connect Source connector for more detailed information.

Reviewer: Examples of snapshotted connector states include partition offsets and any metadata required for a restart or for recovery. If there is a connector failure, recovery behavior depends on.....

> == Parallelism and reconfigurations

Reviewer: initial cap on Reconfigurations

> Kafka Connect Source uses `tasks.max` configuration property to determine parallelism.

Reviewer: add the after uses: Kafka Connect Source uses the `task.max`....

> Even if pipeline's total parallelism is higher than `tasks.max` value, the `tasks.max` value takes precedence.

Reviewer: Reverse: The `tasks.max` value takes precedence, even when a pipeline's total parallelism is higher than that value.

> It is prohibited to have pipeline's total parallelism (edge's local parallelism * number of nodes) lower than `tasks.max`.

Reviewer: A pipeline's total parallelism cannot be lower than the `tasks.max` value.

I'd suggest putting the calculation under the sentence on line 127, along the lines of:

A pipleine's total parallelism is calculated as follows:

    local parallelism of edge * number of nodes

> After creation of an instance of a SourceConnector, Hazelcast Jet asks it to provide `tasks.max` task configurations. If the

Reviewer: Hazelcast Jet requests the `task.max` task configuration when an instance of a Source Connector has been created, .

> SourceConnector returns less than `tasks.max` configs, the remaining processors will become no-op.

Reviewer: If the Source Connector returns ????, the remaining processors are ignored.

???? I wasn't sure what "less than `task.max` configs" meant; is it:

a value lower than the current `tasks.max` configuration

Author: >the remaining processors are ignored.
That wouldn't be true IMHO; processors are not ignored, they are running and waiting for reconfiguration, but main processing method will become no-op (will never return anything unless a reconfiguration is requested).

> I wasn't sure what "less than task.max configs" meant; is it:

Will reword this to:
```
If the SourceConnector returns fewer configurations than `tasks.max` value, the remaining processors will become no-op.
```

> When SourceConnector requests a reconfiguration, Hazelcast asks for a new set of configs and distributes them across the cluster.

Reviewer: When a Source connector requests reconfiguration, Hazelcast requests the new configuration and distributes it across the cluster.

Author: "a new set of configs" is intentional; the source connector returns as many distinct configurations as user requests

Reviewer: And the configuration sets make up the new configuration - in most circumstances, the plural of configuration is configuration. 

Can you clarify what you mean by distinct configurations? Multiple config files/changed lines within a single config file/both of these? That might help pin down the right phrase....

Author: Just for the record (I commented elsewhere): I meant a collection of configurations (set in terms of math), I just went full 3yo with my English and forgot about "configuration set" in other meaning.

> Existing processors will be restarted automatically with new configuration assigned. If new number of returned task configurations

Reviewer: Existing processors automatically restart to use the updated configuration. If supported by the new configuration, previously-ignored processor entries will be processed.

Author: >previously-ignored processor entries will be processed.

There are no processor entries ;) Just processors which were no-op (retuning nothing all the time), that start to produce data

== Fault Tolerance

When you deploy a Kafka Connect Source connector to a Hazelcast cluster, the Jet engine stores snapshots of the connector state.

> Examples of snapshotted connector states include partition offsets and any metadata required for a restart or for recovery. If there is a connector failure, the recovery behavior will depend on the type of connector. Refer to the documentation of your Kafka Connect Source connector for more detailed information.

Reviewer: ```suggestion
Examples of snapshotted connector states include partition offsets and any metadata required for a restart or for recovery. If there is a connector failure, the recovery behavior depends on the type of connector. Refer to the documentation of your Kafka Connect Source connector for more detailed information.
```

== Parallelism and Reconfigurations

Kafka Connect Source uses the `tasks.max` configuration property to determine parallelism.
The `tasks.max` value takes precedence even if pipeline's total parallelism is higher than that value.
A pipeline's total parallelism cannot be lower than `tasks.max`, where total parallelism is calculated as:
```
total parallelism = edge's local parallelism * number of nodes
```

Hazelcast Jet requests `tasks.max` task configurations after an instance of SourceConnector has been created. If the

> SourceConnector returns fewer configurations than `tasks.max` value, the remaining processors will become no-op.

Reviewer: ```suggestion
SourceConnector returns fewer configurations than specified by the `tasks.max` value, the remaining processors do nothing.
```

> When SourceConnector requests a reconfiguration, Hazelcast requests for a new set of configurations and distributes them across the cluster.

Author: I changed it to "the new collection of configuration sets"

I forgot that "configuration set" has already a meaning, while I meant set like in math :D Hopefully now it's ok

> Existing processors automatically restart to use the updated configuration. If new number of returned task configurations

Reviewer: ```suggestion
Existing processors automatically restart and use the updated configuration. If the number of task configuration sets 
```
